### DISCUSSION TITLE
How to iterate over extracted boundary edges?

### DISCUSSION THREAD

**Discussion by utsekaj42** [pyvista/pyvista#4080] [Q&A] [answered]
Maybe I'm missing something, but the result of `extract_feature_edges` does not seem to provide a way of identifying the discrete edges that are extracted? Is there existing functionality to do this somewhere? I have seen some posts and issues about getting the adjacency of points correct, but nothing about dividing the edges in to discrete entities.

https://docs.pyvista.org/examples/01-filter/extract-edges.html#extract-edges-example

https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.extract_feature_edges.html#extract-feature-edges 

In particular, I am taking an annular cross section of an artery and I'd like to get the inner and outer boundary as discrete objects, i.e. if I know my object has two boundaries I'd like to be able to get `edges[0]` and `edges[1]`. As it seems now, I'd have to go through the points of `edges` and determine which boundary they belong to based on some criteria of my own devising (e.g. distance thresholding/clustering).

**Comment by darikg** [comment, marked as answer]
if I understand correctly what you're asking, you can use the connectivity filter to label connected regions in the extracted boundary edges

```python
from pyvista import Disc
mesh = Disc(c_res=20)  # the annulus
boundary = mesh.extract_feature_edges(boundary_edges=True, feature_edges=False, manifold_edges=False)

# The boundary mesh should now have two components, one for the inner and one for the outer
region_id = boundary.connectivity().cell_data['RegionId']
b0 = boundary.extract_cells(region_id == 0)
b1 = boundary.extract_cells(region_id == 1)
```

It's arbitrary which component (b0/b1) is the outer/inner one -- checking the length, e.g. `b0.compute_cell_sizes().cell_data['Length'].sum()`, is probably the easiest way to discriminate them, could also check their winding direction (the outer boundary should be counter-clockwise, the inner clockwise.)
